llmnrd.c

```diff
@@ -150,6 +150,24 @@ static void hostname_change_handle(char *hostname, size_t maxlen)
 	free(newname);
 }
 
+static bool write_pid_file(void)
+{
+	FILE *fp;
+
+	fp = fopen(PIDFILE, "w");
```

Comment:
I think we should open this with `open()` and use `O_CREAT|O_EXCL|O_CLOEXEC` to make sure the file is opened exclusively. The other option would be to keep `fopen` and use `"wx"` mode which achieves the same, but that seems to be a glibc extension and I'd like to keep this portable if possible.

Reply:
OK, seems a good idea to use O_EXCL, I'll cook that up.

Reply:
Done in the updated commit, except that I did not include `O_CLOEXEC` in the set of flags as it is Linux specific, llmnrd is not multithreaded (at least at the point where this function is called) and the file descriptor is closed before exiting the function.

Reply:
Right, not including `O_CLOEXEC` makes perfect sense.